main.js

```diff
@@ -51,65 +50,101 @@ function generateEasyNumbers(numbersWanted) {
 function generateNegativeNumbers(numbersWanted) {
     let allNumbers = [];
 
-    for (let i = 0; i < numbersWanted; i++) {
+    while (allNumbers.length < numbersWanted) {
         let randNumber = Math.floor(Math.random() * 201) - 100;
 
-        if (allNumbers.includes(randNumber)) {
-            i--;
-        } else {
+        if (!allNumbers.includes(randNumber)) {
             allNumbers.push(randNumber);
         }
     }
 
     return allNumbers;
 }
 
-function generateHardNumbers(numbersWanted) {
+function generateHardNumbers(numbersWanted, existingNumbers = []) {
+    const allocatedNumber = getBetween0And9();
+    const totalNumbers = numbersWanted + existingNumbers.length;
     let randNumber = '';
     let allNumbers = [];
-    let numberSelect = Math.floor(Math.random() * 10);
-
-    for (let i = 0; i < numbersWanted; i++) {
-        let rand1in11chance = Math.floor(Math.random() * 11);
-
-        if (rand1in11chance === 10) {
-            randNumber = numberSelect;
-        } else {
-            let randSecondNumber = Math.floor(Math.random() * 10);
 
-            randNumber = parseInt(`${numberSelect}${randSecondNumber}`);
-        }
+    allNumbers = allNumbers.concat(existingNumbers);
```

Comment:
why? all numbers is always empty, why concat an array onto an empty array?

Reply:
`let allNumbers = [...existingNumbers];`